pkg/telemetry/stargate_prometheus_resourcer.go

```diff
@@ -0,0 +1,252 @@
+        replacement: stargate_client_request_latency_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_Latency_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)
+        replacement: stargate_client_request_latency_quantile
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_Failures_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)_total
+        replacement: stargate_client_request_failures_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_Timeouts_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)_total
+        replacement: stargate_client_request_timeouts_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_Unavailables_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)_total
+        replacement: stargate_client_request_unavailables_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_ConditionNotMet_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)
+        replacement: stargate_client_request_condition_not_met_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_UnfinishedCommit_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)
+        replacement: stargate_client_request_unfinished_commit_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+      - regex: persistence_cassandra_(\d_\d+)_org_apache_cassandra_metrics_ClientRequest_ContentionHistogram_(Read|Write|CASRead|CASWrite|RangeSlice|ViewWrite)_count
+        replacement: stargate_client_request_contention_histogran_total
+        sourceLabels:
+        - __name__
+        targetLabel: __name__
+`
+
+var stargateEndpointHolder promapi.ServiceMonitor
+
+func init() {
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	_, _, err := decode([]byte(stargateEndpointString), nil, &stargateEndpointHolder)
+	if err != nil {
+		fmt.Println("Fatal error initialising EndpointHolder in pks/telemetry/stargate_prometheus_resourcer.go", err)
+		os.Exit(1)
+	}
+}
+
+// mustLabels() returns the set of labels essential to managing the Prometheus resources. These should not be overwritten by the user.
+func (cfg StargatePrometheusResourcer) mustLabels() map[string]string {
+	return map[string]string{
+		k8ssandraapi.ManagedByLabel:  k8ssandraapi.NameLabelValue,
+		k8ssandraapi.PartOfLabel:     k8ssandraapi.PartOfLabelValue,
+		k8ssandraapi.DatacenterLabel: cfg.StargateName,
```

Comment:
Not sure `StargateName` makes sense as the value for `DatacenterLabel`.

Reply:
Good catch. Not sure what I was thinking there.